Marlin/src/module/temperature.cpp

```diff
@@ -1219,17 +1219,23 @@ void Temperature::manage_heater() {
 
   void Temperature::log_user_thermistor(const CustomThermistorIndex t_index, const bool eprom/*=false*/) {
 
+    char str[16];
+
     SERIAL_ECHO_START();
     if (eprom) SERIAL_ECHOPGM("  M305 ");
     SERIAL_CHAR('P');
     SERIAL_CHAR('0' + t_index);
 
     const user_thermistor_t &t = user_thermistor[t_index];
 
-    SERIAL_PRINTF(" R%0.1f", t.series_res);
-    SERIAL_PRINTF(" T%0.1f", t.res_25);
-    SERIAL_PRINTF(" B%0.1f", t.beta);
-    SERIAL_PRINTF(" C%0.9f", t.sh_c_coeff);
+    sprintf_P(str, PSTR(" R%0.1f"), t.series_res); SERIAL_ECHO(str);
+    sprintf_P(str, PSTR(" T%0.1f"), t.res_25); SERIAL_ECHO(str);
+    sprintf_P(str, PSTR(" B%0.1f"), t.beta); SERIAL_ECHO(str);
+    sprintf_P(str, PSTR(" C%0.9f"), t.sh_c_coeff); SERIAL_ECHO(str);
+//    SERIAL_PRINTF(" R%0.1f", t.series_res);
+//    SERIAL_PRINTF(" T%0.1f", t.res_25);
+//    SERIAL_PRINTF(" B%0.1f", t.beta);
+//    SERIAL_PRINTF(" C%0.9f", t.sh_c_coeff);
```

Comment:
NOTE: The AVR version of `printf` has no support for `float` so we have to use other methods.

Reply:
right, Okey doke. Well, you know Marlin way way better than me to fix that.

Reply:
Someday we might even document all this stuff!